From 190b89f404520a01512196fa63497b46ca75fb01 Mon Sep 17 00:00:00 2001
From: Gabor Szadovszky <gabor.szadovszky@cloudera.com>
Date: Mon, 22 Aug 2016 09:53:41 -0700
Subject: [PATCH 0735/1164] CDH-43290: HIVE-14588: Add S3 credentials to the
 hidden configuration variable supported on
 HIVE-14207 (Gabor Szadovszky, reviewed by Sergio
 Pena)

(cherry picked from commit 39c858c77d94d12b5d1d11dd780587c031b49c1d)

Change-Id: I449650d09c9e2cec5db61080355baf5ae9b7ae80
---
 .../java/org/apache/hadoop/hive/conf/HiveConf.java |   10 +++++++++-
 1 file changed, 9 insertions(+), 1 deletion(-)

diff --git a/common/src/java/org/apache/hadoop/hive/conf/HiveConf.java b/common/src/java/org/apache/hadoop/hive/conf/HiveConf.java
index e798758..5e7d1a9 100644
--- a/common/src/java/org/apache/hadoop/hive/conf/HiveConf.java
+++ b/common/src/java/org/apache/hadoop/hive/conf/HiveConf.java
@@ -1934,7 +1934,15 @@ public void setSparkConfigUpdated(boolean isSparkConfigUpdated) {
         "hive.security.authenticator.manager,hive.security.authorization.manager,hive.users.in.admin.role",
         "Comma separated list of configuration options which are immutable at runtime"),
     HIVE_CONF_HIDDEN_LIST("hive.conf.hidden.list",
-        METASTOREPWD.varname + "," + HIVE_SERVER2_SSL_KEYSTORE_PASSWORD.varname,
+        METASTOREPWD.varname + "," + HIVE_SERVER2_SSL_KEYSTORE_PASSWORD.varname
+        // Adding the S3 credentials from Hadoop config to be hidden
+        + ",fs.s3.awsAccessKeyId"
+        + ",fs.s3.awsSecretAccessKey"
+        + ",fs.s3n.awsAccessKeyId"
+        + ",fs.s3n.awsSecretAccessKey"
+        + ",fs.s3a.access.key"
+        + ",fs.s3a.secret.key"
+        + ",fs.s3a.proxy.password",
         "Comma separated list of configuration options which should not be read by normal user like passwords"),
     HIVE_MOVE_FILES_THREAD_COUNT("hive.mv.files.thread", 25, "Number of threads"
          + " used to move files in move task. Set it to 0 to disable multi-threaded file moves. This parameter is also used by"
-- 
1.7.9.5

